### ISSUE TITLE
HA 0.88 brings a 'new' / 'better'? custom component layout...

### ISSUE THREAD

**Issue by Swiftnesses** [zxdavb/evohome#29]
`Integrations need to be in their own folder. Change climate/evohome_cc.py to evohome_cc/climate.py. This will stop working soon.`

**Comment by zxdavb** [comment]
I will fix this (and a few other issues) this weekend.

**Comment by Swiftnesses** [comment]
Thanks :)

FYI, it also spits out errors like:

```
Error loading custom_components.evohome_cc.climate. Make sure all dependencies are installed
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/homeassistant/loader.py", line 147, in _load_file
    module = importlib.import_module(path)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 962, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'custom_components.evohome_cc.climate'; 'custom_components.evohome_cc' is not a package
```

**Comment by zxdavb** [comment]
feedback to me after the weekend if it is still there...

**Comment by zxdavb** [comment]
I believe this has been 'fixed'. However, it doesn't work with HA v0.88b1 due to stuff missing from `homeassistant/components/climate/const.py`.

It _does_ work with HA v0.89-dev (as at 16/02/2019), and may well work with HA v0.88 when it is released.

Don't bother attempting to use this code, unless you're running HA v0.88b2 or later.

**Comment by Swiftnesses** [comment]
I'm using 0.88B2 sadly and get the error about missing dependencies, guess I need to wait now for HA to sort the issue...

**Comment by Beertje135** [comment]
Same problem after update to 0.88.0. Missing dependencies.

**Comment by zxdavb** [comment]
I've had confirmation that the necessary change ruby be there until 0.89.

Until then, just use an older version of the component.

**Comment by zxdavb** [comment]
@Swiftnesses your first and second posts are due to different errors.

The first is survivable (i.e. the component will run ok), the second is not.

The second is because you're using master, and HA <0.89

**Comment by megapearl** [comment]
I'm using homeassistant v0.88.0 and have it working by changing to the right repo using:

```
root@ubuntu-server:/etc/homeassistant/custom_components/evohome_cc# git checkout integrated-folders
Branch 'integrated-folders' set up to track remote branch 'integrated-folders' from 'origin'.
Switched to a new branch 'integrated-folders'
root@ubuntu-server:/etc/homeassistant/custom_components/evohome_cc# git pull
Already up to date.
```
Changed the custom_components.json file so the contents are using the new layout:

```
root@ubuntu-server:/etc/homeassistant/custom_components/evohome_cc# cat custom_components.json
{
    "evohome_cc": {
        "version":         "0.9.5",
        "local_location":  "custom_components/evohome_cc/__init__.py",
        "remote_location": "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/__init__.py",
        "changelog":       "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/CHANGELOG.md",
        "visit_repo":      "https://github.com/zxdavb/evohome"
    },
    "climate.evohome_cc": {
        "version":         "0.9.5",
        "local_location":  "custom_components/evohome_cc/climate.py",
        "remote_location": "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/climate.py",
        "changelog":       "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/CHANGELOG.md",
        "visit_repo":      "https://github.com/zxdavb/evohome"
    },
    "water_heater.evohome_cc": {
        "version":         "0.9.5",
        "local_location":  "custom_components/evohome_cc/water_heater.py",
        "remote_location": "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/water_heater.py",
        "changelog":       "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/CHANGELOG.md",
        "visit_repo":      "https://github.com/zxdavb/evohome"
    }
}

```

In configuration.yaml

```
evohome_cc:
    username: !secret evohome_username
    password: !secret evohome_password
    scan_interval: 360
    high_precision: true
    use_heuristics: true
```

Regards,
Donald.

**Comment by Stephan296** [comment]
> I'm using homeassistant v0.88.0 and have it working by changing to the right repo using:
> 
> ```
> root@ubuntu-server:/etc/homeassistant/custom_components/evohome_cc# git checkout integrated-folders
> Branch 'integrated-folders' set up to track remote branch 'integrated-folders' from 'origin'.
> Switched to a new branch 'integrated-folders'
> root@ubuntu-server:/etc/homeassistant/custom_components/evohome_cc# git pull
> Already up to date.
> ```
> Changed the custom_components.json file so the contents are using the new layout:
> 
> ```
> root@ubuntu-server:/etc/homeassistant/custom_components/evohome_cc# cat custom_components.json
> {
>     "evohome_cc": {
>         "version":         "0.9.5",
>         "local_location":  "custom_components/evohome_cc/__init__.py",
>         "remote_location": "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/__init__.py",
>         "changelog":       "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/CHANGELOG.md",
>         "visit_repo":      "https://github.com/zxdavb/evohome"
>     },
>     "climate.evohome_cc": {
>         "version":         "0.9.5",
>         "local_location":  "custom_components/evohome_cc/climate.py",
>         "remote_location": "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/climate.py",
>         "changelog":       "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/CHANGELOG.md",
>         "visit_repo":      "https://github.com/zxdavb/evohome"
>     },
>     "water_heater.evohome_cc": {
>         "version":         "0.9.5",
>         "local_location":  "custom_components/evohome_cc/water_heater.py",
>         "remote_location": "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/water_heater.py",
>         "changelog":       "https://raw.githubusercontent.com/zxdavb/evohome/integrated-folders/CHANGELOG.md",
>         "visit_repo":      "https://github.com/zxdavb/evohome"
>     }
> }
> ```
> In configuration.yaml
> 
> ```
> evohome_cc:
>     username: !secret evohome_username
>     password: !secret evohome_password
>     scan_interval: 360
>     high_precision: true
>     use_heuristics: true
> 
> custom_updater:
>   component_urls:
>     - https://raw.githubusercontent.com/zxdavb/evohome/evohome_cc/custom_components.json
> ```
> Regards,
> Donald.

With this "config" it doesn't track the evohome component in custom_updater.
Did it show's the evohome custom component at your tracker card?

**Comment by zxdavb** [comment]
This has been addressed with release tagged at **v0.9.9** (on the **master** branch).
